ingest/scripts/split-cdsCoverage-columns.py

```diff
@@ -0,0 +1,104 @@
+#! /usr/bin/env python
+import argparse
+import pandas as pd
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description="Split up the aggregated cdsCoverage column from nextclade.tsv into gene_coverage columns",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter
+    )
+    parser.add_argument(
+        "--metadata",
+        required=True,
+        help="Path to the metadata.tsv or nextclade.tsv file"
+    )
+    parser.add_argument(
+      "--genes",
+      required=True,
+      help="Comma-separated list of genes (e.g., '3CLpro,NTPase') to convert to coverage"
+    )
+    parser.add_argument(
+      "--cdsCoverage",
+      default="cdsCoverage",
+      help="Column with gene coverage string"
+    )
+    parser.add_argument(
+      "--round",
+      type=int,
+      default=None,
+      help="If defined, round to this number of decimal places"
+    )
+    parser.add_argument(
+      "--chunksize",
+      type=int,
+      default=1000,
+      help="Chunk size for reading TSV"
+    )
+    parser.add_argument(
+      "--output",
+      default="coverages.tsv",
+      help="Output TSV filename where gene_coverage columns are appended to input TSV file"
+    )
+    return parser.parse_args()
+
+
+def parse_coverage_field(field_value, genes, round_digits=None):
+    """Convert a coverage string like '3CLpro:1,NTPase:0.99' into a dict for selected genes."""
+    gene_dict = {}
+
+    # Set gene order and default to 0.0
+    for gene in genes:
+        gene_dict[f"{gene}_coverage"] = float(0.0)
+
+    for item in field_value.split(","):
+        if ":" not in item:
+            continue
+        gene, val = item.split(":")
+        # Set recognized genes to value
+        if gene in genes:
+            try:
+                value = float(val)
+
+                # Optionally round the value, especially for excell which may auto convert long float values to text
+                if round_digits is not None:
+                    value = round(value, round_digits)
+
+                # Add original or rounded value to dictionary
+                gene_dict[f"{gene}_coverage"] = value
+            except ValueError:
+                gene_dict[f"{gene}_coverage"] = None
+
+    return gene_dict
+
+def process_chunk(chunk, cds_coverage_col, genes, round_digits=None):
+    """For each row, parse gene coverages and append as new columns."""
+    coverage_dicts = []
+
+    for _, row in chunk.iterrows():
```

Comment:
I'm generally wary of `iterrows` because it can be slow, but won't block on this here since the ingest workflow still completes pretty quickly.

I do wonder if things would be simpler here _without_ pandas, i.e. read TSV with csv.DictReader, loop over rows to add gene coverage rows, and then output with csv.DictWriter. That would remove the complexity of chunks, converting between dict and DataFrame, and concatenating DataFrame.

Reply:
Thanks for refactoring this without pandas! I left some small comments, but this looks good to me.